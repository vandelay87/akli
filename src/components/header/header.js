import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { robotoRegular } from '../../styles/fonts';
import { color } from '../../styles/colors';

const Header = ({ title }) => (
  <StyledHeader>
    <StyledTitle>{title}</StyledTitle>
  </StyledHeader>
);

const StyledHeader = styled.header`
  background: ${color.primary};
  padding: 16px;
`;
const StyledTitle = styled.span`
  ${robotoRegular}
  color: ${color.onPrimary};
  font-size: 1.25rem;;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  font-weight: 500;
  line-height: 2rem;
  letter-spacing: .0125em;
  margin-left: 16px;
`;

Header.propTypes = {
  title: PropTypes.string.isRequired,
};

export default Header;
