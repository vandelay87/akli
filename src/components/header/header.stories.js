import React from 'react';
import { text } from '@storybook/addon-knobs';
import Header from './header';

export default {
  component: Header,
  title: 'Header',
};

export const header = () => {
  const knob = text('Title', 'My Website');

  return <Header title={knob} />;
};
