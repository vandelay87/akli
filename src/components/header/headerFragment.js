import { graphql } from 'gatsby'

export const query = graphql`
  fragment Header on ContentfulWebsiteSettings {
    header {
      title
    }
  }
`
