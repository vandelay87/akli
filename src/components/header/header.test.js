import renderer from 'react-test-renderer'
import { header } from './header.stories'

describe('header.js', () => {
  it('renders component', () => {
    const tree = renderer.create(header()).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
