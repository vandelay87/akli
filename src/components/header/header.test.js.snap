// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header.js renders component 1`] = `
<header
  className="sc-bdVaJa hkQuXJ"
>
  <span
    className="sc-bwzfXH cJuWJO"
  >
    My Website
  </span>
</header>
`;
